galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/routes/PropertyQueryRoute.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.framework.api.cps.internal.routes;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import dev.galasa.framework.api.common.InternalServletException;
+import dev.galasa.framework.api.common.QueryParameters;
+import dev.galasa.framework.api.common.ResponseBuilder;
+import dev.galasa.framework.api.common.ServletError;
+import dev.galasa.framework.spi.ConfigurationPropertyStoreException;
+import dev.galasa.framework.spi.FrameworkException;
+import dev.galasa.framework.spi.IFramework;
+
+import static dev.galasa.framework.api.common.ServletErrorMessage.*;
+
+public class PropertyQueryRoute extends CPSRoute{
+
+    private static final String path = "/cps/(.*)/properties([?]?|[^/])+$";
+
+    public PropertyQueryRoute(ResponseBuilder responseBuilder, IFramework framework) {
+        super(responseBuilder, path , framework);
+    }
+
+   
+    @Override
+    public HttpServletResponse handleRequest(String pathInfo, QueryParameters queryParams,HttpServletRequest req, HttpServletResponse response)
+            throws ServletException, IOException, FrameworkException {
+        String namespace = getNamespaceFromURL(pathInfo);
+        String properties = getNamespaceProperties(namespace, queryParams);
+        return getResponseBuilder().buildResponse(response, "application/json", properties, HttpServletResponse.SC_OK); 
+    }
+
+    private String getNamespaceProperties(String namespace, QueryParameters queryParams) throws FrameworkException{
+        String properties = "";
+         try {
+            nameValidator.assertNamespaceCharPatternIsValid(namespace);
+            if (hiddenNameSpaces.contains(namespace)) {
+            ServletError error = new ServletError(GAL5016_CPS_HIDDEN_NAMESPACE_ERROR, namespace);
+			throw new InternalServletException(error, HttpServletResponse.SC_NOT_FOUND);
+            }
+            String prefix = queryParams.getSingleString("prefix", null);
+            String suffix = queryParams.getSingleString("suffix", null);
+            properties = getProperties(namespace, prefix, suffix);
+        }catch (FrameworkException f){
+            if (f instanceof InternalServletException){
+                throw f;
```

Comment:
Not 100% sure about the instanceof - might be worth catching the `InternalServletException` and then having a separate catch for `FrameworkException`

Reply:
Ah Mike raised this as well! Resolving.